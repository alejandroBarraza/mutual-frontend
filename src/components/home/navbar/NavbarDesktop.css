.wrapper {
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.2);
  font-family: "Public Sans", sans-serif;
  color: #71808c;
  border-radius: 0.5rem;
  width: 100%;
  height: 14vh;
  padding: 1rem;
}

.navbar {
  height: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  font-size: calc(0.8rem + 1vmin);
}
.navbar-items {
  display: flex;
  align-items: center;
  justify-content: space-between;
  text-decoration: none;
  font-size: calc(0.4rem + 1vmin);
  list-style: none;
  color: #71808c;
  transition: all 0.2s ease;
}

.navbar-items li {
  padding-left: 0rem;
  margin-left: 2rem;
}
.navbar-items li a:hover {
  color: #bed700;
}

/* .logo {
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  height: 100%;
} */

image {
  height: 100%;
}
