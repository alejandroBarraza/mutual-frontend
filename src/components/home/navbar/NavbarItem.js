export const NavbarItem = [
    {
        title: 'Inicio',
        url: '/',
        cName: 'navbar-items'
    },
    {
        title: 'Noticias',
        url: '#',
        cName: 'navbar-items',
        section: 'news'
    },
    // {
    //     title: 'Casos',
    //     url: '#',
    //     cName: 'navbar-items',
    // },
    {
        title: 'Descargables',
        url: '/descargables',
        cName: 'navbar-items'
    },
    {
        title: 'Contactanos',
        url: '/form',
        cName: 'navbar-items'
    },
];
