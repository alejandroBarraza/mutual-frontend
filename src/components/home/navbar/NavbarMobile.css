@import url('https://fonts.googleapis.com/css2?family=Public+Sans:wght@300;400&display=swap');
.wrapper-mobile {
    box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.2);
    font-family: 'Public Sans', sans-serif;
    color: #71808c;
    border-radius: 0.5rem;
}
.navbar-mobile {
    height: 8vh;
    display: flex;
    align-items: center;
    justify-content: space-between;
    position: relative;
}
.logo-mobile {
    height: 48px;
    width: 48px;
    display: flex;
    justify-content: center;
    align-items: center;
}

img {
    max-width: 100%;
    max-height: 100%;
}
.navbar-items {
    position: relative;
}
/* .menu {
    display: flex;
    flex-direction: column;
    list-style: none;
    width: 100%;
    height: 100%;
    left: -100%;
    opacity: 1;
    transition: all 0.5s ease;
} */

.navbar-mobile-menu {
    height: 92vh;
    display: flex;
    flex: 1;
    flex-direction: column;
    width: 100%;
    overflow: auto;
    position: absolute;
    top: 8vh;
    left: -100%;
    align-items: center;
    text-align: center;
    list-style-type: none;
    transition: all 0.5s ease;
    opacity: 1;
    backdrop-filter: blur(10px);
    -webkit-backdrop-filter: blur(10px);
}
.navbar-mobile-menu.active {
    left: 0;
    transition: all 0.5s ease;
}
.navbar-mobile-menu li {
    padding: 2rem;
    width: 100%;
    text-align: center;
}

.navbar-mobile-menu li a {
    text-decoration: none;
    color: #71808c;
}
.navbar-mobile-menu li a:hover {
    text-decoration: none;
    color: #bed700;
    transition: all 0.1s ease-in-out;
}
