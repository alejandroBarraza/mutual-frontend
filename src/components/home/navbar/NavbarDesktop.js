import React from 'react';
import { Link } from 'react-router-dom';
import { NavbarItem } from './NavbarItem';
import './NavbarDesktop.css';
import '../../../container.css';
import { Link as Lonk} from "react-scroll";


export const NavbarbarDesktop = () => {
    return (
        <div className='wrapper'>
            <nav className='navbar container'>
                <Link className='link' to='/'>
                    <img className='image' src='/mutual-svg.svg' alt='logo-mutual' />
                </Link>
                <ul className='navbar-items'>
                    {NavbarItem.map((item, index) => (
                        item.section ?
                            <li key={index}>
                                <Lonk
                                    activeClass="active"
                                    to={item.section}
                                    spy={true}
                                    smooth={true}
                                    offset={-70}
                                    duration={500}>
                                    <Link className={item.cName} to={item.url}>
                                        {item.title}
                                    </Link>
                                </Lonk>

                            </li>
                            :
                            <li key={index}>
                                <Link className={item.cName} to={item.url}>
                                    {item.title}
                                </Link>
                            </li>
                    ))}
                </ul>
            </nav>
        </div>
    );
};
