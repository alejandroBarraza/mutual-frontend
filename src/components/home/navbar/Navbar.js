import React from 'react';
import { useViewport } from '../../hooks/useViewport';
import { NavbarbarDesktop } from './NavbarDesktop';
import { NavbarMobile } from './NavbarMobile';

export const Navbar = () => {
    const { width } = useViewport();
    const breakpoint = 768;
    return width > breakpoint ? <NavbarbarDesktop /> : <NavbarMobile />;
};
