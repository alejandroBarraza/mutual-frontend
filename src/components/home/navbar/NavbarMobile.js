import React, { useState } from 'react';
import MenuRoundedIcon from '@mui/icons-material/MenuRounded';
import MenuOpenRoundedIcon from '@mui/icons-material/MenuOpenRounded';
import { Link as Lonk, animateScroll as scroll } from "react-scroll";
import '../../../container.css';
import './NavbarMobile.css';
import { NavbarItem } from './NavbarItem';
import { Link } from 'react-router-dom';
export const NavbarMobile = () => {
    const [navClicked, setnavClicked] = useState(!false);
    const handleClick = () => setnavClicked(!navClicked);

    return (
        <div className='wrapper-mobile'>
            <nav className=' navbar-mobile container'>
                <div className='logo-mobile'>
                    <Link className='link' to='/'>
                        <img className='image' src='/mutual-svg.svg' alt='logo-mutual' onClick={() => { setnavClicked(true) }} />
                    </Link>
                </div>
                <div onClick={handleClick}>
                    {navClicked ? (
                        <MenuRoundedIcon style={{ cursor: 'pointer' }} />
                    ) : (
                        <MenuOpenRoundedIcon style={{ cursor: 'pointer' }} />
                    )}
                </div>
            </nav>
            <ul className={navClicked ? 'navbar-mobile-menu' : 'navbar-mobile-menu active'}>
                {NavbarItem.map((item, index) => (
                    <li key={index}>
                        {item.section != null ?
                            <Lonk
                                activeClass="active"
                                to={item.section}
                                spy={true}
                                smooth={true}
                                offset={-70}
                                duration={500}
                                onClick={handleClick}>
                                <a href={item.url}>{item.title}</a>
                            </Lonk> :
                            <a href={item.url}>{item.title}</a>
                        }
                    </li>
                ))}
            </ul>
        </div>
    );
};
