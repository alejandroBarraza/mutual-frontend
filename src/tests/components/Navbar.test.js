import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';

import { render, cleanup } from '@testing-library/react';

import { BrowserRouter as Router } from 'react-router-dom';

import { NavbarbarDesktop } from '../../components/home/navbar/NavbarDesktop';
import { NavbarMobile } from '../../components/home/navbar/NavbarMobile';

afterEach(cleanup);

describe('testing <Navbar/> component', () => {
    it('should check render component at breakpoint 768px', async () => {
        const width = (window.innerWidth = 1200);
        const breakpoint = 768;
        const { container: containerMobile } = render(
            <Router>
                <NavbarMobile />
            </Router>
        );
        const { container: containerDekstop } = render(
            <Router>
                <NavbarbarDesktop />
            </Router>
        );
        const navbarDeskstopClass = containerDekstop.firstChild.classList.contains('wrapper');
        const navbarMobileClass = containerMobile.classList.contains('wrapper-mobile');

        if (width > breakpoint) {
            expect(navbarDeskstopClass).toBe(true);
        } else {
            expect(navbarMobileClass).toBe(true);
        }
    });
    it('should match snapshot', () => {
        const { container } = render(
            <Router>
                <NavbarbarDesktop />
            </Router>
        );
        expect(container).toMatchSnapshot();
    });
});
