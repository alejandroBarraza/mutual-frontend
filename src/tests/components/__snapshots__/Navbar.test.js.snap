// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <Navbar/> component should match snapshot 1`] = `
<div>
  <div
    class="wrapper"
  >
    <nav
      class="navbar container"
    >
      <a
        class="link"
        href="/"
      >
        <img
          alt="logo-mutual"
          class="image"
          src="/mutual-svg.svg"
        />
      </a>
      <ul
        class="navbar-items"
      >
        <li>
          <a
            class="navbar-items"
            href="/"
          >
            Inicio
          </a>
        </li>
        <li>
          <a>
            <a
              class="navbar-items"
              href="/"
            >
              Noticias
            </a>
          </a>
        </li>
        <li>
          <a
            class="navbar-items"
            href="/descargables"
          >
            Descargables
          </a>
        </li>
        <li>
          <a
            class="navbar-items"
            href="/form"
          >
            Contactanos
          </a>
        </li>
      </ul>
      <button
        aria-label="Iniciar SecciÃ³n"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element="true"
        tabindex="0"
        type="button"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
          href="https://aunteespero-backend.herokuapp.com/admin"
          rel="noopener"
          target="_blank"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1kttvr3-MuiSvgIcon-root"
            data-testid="AccountCircleIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
            />
          </svg>
        </a>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </nav>
  </div>
</div>
`;
